session/session_test.go

```diff
@@ -4264,3 +4266,62 @@ func (s *testSessionSuite3) TestGlobalTemporaryTable(c *C) {
 	// The global temporary table data is discard after the transaction commit.
 	tk.MustQuery("select * from g_tmp").Check(testkit.Rows())
 }
+
+type testTxnStateSuite struct {
```

Comment:
Besides checking the txnInfo directly, I think you also need to test getting the transaction's information by querying the TIDB_TRX table.